### PR TITLE
[release-1.9] :bug: Bumping knative.dev/hack to latest main

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- [release-1.9] :bug: Bumping knative.dev/hack to latest main
  - In this PR I've manually bumped the knative.dev/hack to main, before cherry-picking the https://github.com/knative/hack/pull/268 to `release-1.9`, so that regular update-deps will consume it.

/kind bug

Fixes https://github.com/knative-sandbox/homebrew-kn-plugins/issues/134

**Release Note**

```release-note
Fixing the invalid checksums on MacOS platform, for release-1.9, making homebrew install work again.
```

### CODE REVIEW

**Comment by cardil** [comment]
@upodroid here's my plan:

In this PR I've manually bumped the knative.dev/hack to latest main. After this PR lands, and we do a dot-release for kn-event, and confirm all signatures and checksums are calculated properly, we could go ahead with cherry-picking the https://github.com/knative/hack/pull/268 to `release-1.9` branch. After that, we can fix all other affected repos (with update-deps PR and a dot-release).

**Review by upodroid — APPROVED** [review]
LGTM

/hold

Merge this after verifying the nightly as described in #270

**Pull request merged**
